### ISSUE TITLE
ipmag.vgpmag_magic erases 'normal' poles if there are reverse ones

### ISSUE THREAD

**Issue by ltauxe** [PmagPy/PmagPy#649]
**What command was run:**

ipmag.vgpmap_magic(dir_path=dir_path,size=50,flip=True,save_plots=True,lat_0=60,rsym='b^',rsize=50,fmt='png')

and

ipmag.vgpmap_magic(dir_path=dir_path,size=50,flip=False,save_plots=True,lat_0=60,rsym='b^',rsize=50,fmt='png')

**What happened:**

There are no 'normal' poles when flip is set to True.  This used to work but something happened!  

**What I expected to happen:**

I expect the red and blue symbols as advertized for when flip is set to True

**Data file(s):**
[sites.txt](https://github.com/PmagPy/PmagPy/files/8484518/sites.txt)

**Screenshot(s):**
plot with both normal and reverse data only shows only reverse poles:

![Golan-Heights_VGP_map](https://user-images.githubusercontent.com/174861/163265498-67554858-8f4e-4d98-a7f2-d129367fc23d.png)

without flip, you get only the normal poles (as expected).  

![Golan-Heights_VGP_map](https://user-images.githubusercontent.com/174861/163265773-b39cab0b-4a30-40cb-a9b5-28af74b9b1ae.png)

The way it is supposed to work (and does on jupyterhub.earthref.org, weirdly, with the same version of PmagPy).

![Screen Shot 2022-04-13 at 1 36 38 PM](https://user-images.githubusercontent.com/174861/163266195-730ddb0d-77d0-4492-8f07-c342cba2967e.png)

**PmagPy version:**
pmagpy-4.2.98

**Comment by Swanson-Hysell** [comment]
It looks like the code is first plotting the normal:

```
    if len(lats) > 0:
        # add the lats and lons of the poles
        pmagplotlib.plot_map(FIG['map'], lats, lons, Opts)
```

and then plotting the reversed:

```
    if len(rlats) > 0:
        Opts['sym'] = rsym
        Opts['symsize'] = rsize
        # add the lats and lons of the poles
        pmagplotlib.plot_map(FIG['map'], rlats, rlons, Opts)
```

What might be happening is that the land and ocean in the second ```pmagplotlib.plot_map()``` is covering up the first set of points.

**Comment by ltauxe** [comment]
yeah but the flag ‘pltgrid’ in Opts is turned off, so it isn’t supposed to do that.  I tried a bunch of things and couldn’t get it to work….  

lisa

**Comment by ltauxe** [comment]
and the really weird thing is that the exact same code works perfectly well on the jupyterhub site.  maybe I should just move on….  

lisa

**Comment by Swanson-Hysell** [comment]
An inelegant solution that works is where:

```
    if len(rlats) > 0:
        Opts['sym'] = rsym
        Opts['symsize'] = rsize
        # add the lats and lons of the poles
        pmagplotlib.plot_map(FIG['map'], rlats, rlons, Opts)
```

is replaced with:

```
    if len(rlats) > 0:
        Opts['sym'] = rsym
        Opts['symsize'] = rsize
        if len(lats) > 0:
            color, symbol = Opts['sym'][0], Opts['sym'][1]
            plt.gca().scatter(rlons, rlats, s=Opts['symsize'], c=color, marker=symbol,
                              transform=ccrs.PlateCarree(), edgecolors=Opts['edgecolor'])
        else:
            pmagplotlib.plot_map(FIG['map'], rlats, rlons, Opts)
```

**Comment by Swanson-Hysell** [comment]
That solution just plots with scatter on the already generated axis and gives:
<img width="598" alt="Screen Shot 2022-04-13 at 2 26 30 PM" src="https://user-images.githubusercontent.com/4332322/163273243-bfc0f70b-63ea-431d-a4a1-1e55b441bf6f.png">

**Comment by Swanson-Hysell** [comment]
> yeah but the flag ‘pltgrid’ in Opts is turned off, so it isn’t supposed to do that. I tried a bunch of things and couldn’t get it to work…. lisa
> […](#)

I think it is the plotting of the oceans and land rather than the grid that is hiding the normal points when the reversed map is generated with pmagplotlib.plot_map().

It is likely preferable to be using the map that is already generated rather than making a whole new map (which is what the edited code above does). I can commit that if it looks ok to you.

**Comment by ltauxe** [comment]
yes please.  that whole function needs re-writing I guess - but not today!  

lisa

**Comment by Swanson-Hysell** [comment]
Did commit https://github.com/PmagPy/PmagPy/commit/e2a7b00060a939d5b45224fd9eaad53dc4bb2d6f solve the issue? If so, we can go ahead and close.

**Comment by ltauxe** [comment]
no it didn’t.  sorry.

**Comment by apivarunas** [comment]
This works as advertised with pmagpy==4.2.124.

`ipmag.vgpmap_magic(dir_path=dir_path,size=50,flip=True,save_plots=True,lat_0=60,rsym='b^',rsize=50,fmt='png')`

![Image](https://github.com/user-attachments/assets/1e4cbaa7-f145-40ab-8f1f-b64da0cad401)
